sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/AiOperationNameSpanProcessor.java

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.monitor.opentelemetry.exporter;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.trace.ReadWriteSpan;
+import io.opentelemetry.sdk.trace.ReadableSpan;
+import io.opentelemetry.sdk.trace.SpanProcessor;
+
+/**
+ * SpanProcessor implementation to update operation name.
+ */
+// note: operation name for requests is handled during export so that it can use the updated span name from routing instrumentation
+//       if we (only) set operation name on requests here, it would be based on span name at startSpan
+public class AiOperationNameSpanProcessor implements SpanProcessor {
```

Comment:
since we haven't thought through the public API for this yet, I think we can leave it as package protected for now, then we can follow up in a separate PR with public API proposal

```suggestion
class AiOperationNameSpanProcessor implements SpanProcessor {
```